Add Jira styles

ackernaut opened the pull request:
In support of #2773 

- [x] Updates styles for menus including card integration options and search
- [x] Updates styles for panels/rows including provider list and details under team settings
- [x] Includes housekeeping to reduce usage of `ui.js`
- [x] Updates buttons/panels for organization, notifications, and user profile views

### Test
Everything looks as expected per latest designs/theme:
- [ ] Add provider via card menu
- [ ] Integrate a card via menu
- [ ] Interact with provider list under team settings
- [ ] Interact with provider details under team settings
- [ ] Interact with notifications
- [ ] Interact with organization billing and members
- [ ] Interact with user profile

ackernaut commented:
I think this is ready for your eyes @mattkrick 

There is one nuance that I found that I tested with and without style changes and it seems to be there either way. If a user blurs the menu search control for integrations they can’t focus on the input again. It’s tricky to do but I did it to test styling of the default border color. If you get a menu with only a few items, there is the specified gutter (8px) after the last item. You can click there to blur the input.

mattkrick commented on src/universal/components/AddToGitHubMenuItem.tsx:
```diff
@@ -24,7 +24,7 @@ const AddToGitHubMenuItem = forwardRef((props: Props, ref) => {
         <MenuItemLabel>
           <MenuItemComponentAvatar>
-            <GitHubSVG />
+            <GitHubSVG18 />
```
Why are we constraining a DOM SVG to a particular size? We should instead style the svg like any other element

mattkrick commented on src/universal/components/Panel/Panel.js:
```diff
@@ -1,52 +1,43 @@
 
 const PanelControls = styled('div')(({compact}) => {
-  const padding = compact ? ui.panelCompactGutter : ui.panelGutter
+  const padding = compact ? ROW_GUTTER_COMPACT : ROW_GUTTER
```
looks like the compact variable isn't used. remove?

mattkrick commented on src/universal/components/Row/RowInfoCopy.ts:
```diff
@@ -0,0 +1,8 @@
+import {ROW_SUBHEADING_STYLES} from 'universal/styles/rows'
+
+const RowInfoCopy = styled('div')({
+  ...ROW_SUBHEADING_STYLES
```
lets keep those styles inside this component & then if another component needs to based off of this one we can use `RowInfoCopy.withComponent`

mattkrick commented on src/universal/components/SuggestedIntegrationGitHubMenuItem.tsx:
```diff
@@ -20,6 +21,12 @@ interface Props {
   query: string
 }
 
+const BlockSVG = styled('div')({
```
this div looks unnecessary, can we apply the styles to MenuItemComponentAvatar?

mattkrick commented on src/universal/modules/teamDashboard/components/ProviderRow/ProviderRowName.tsx:
```diff
@@ -1,64 +1,70 @@
-import React from 'react'
```
delete?

ackernaut replied:
Will do, I think in #2740 we remove `userCount` and `integrationCount` as I’m wanting to simplify the design and not have those icons anymore.

mattkrick commented on src/universal/styles/providers.js:
```diff
@@ -0,0 +1,10 @@
+// for provider UI components
+
+export const GITHUB_NAME = 'GitHub'
```
let's put all these inside `universal/constEnums.d.ts` under an enum like `Provider`.
That way, the compiled code doesn't have to import anything, the references are just replaced by the values

mattkrick commented on src/universal/styles/rows.ts:
```diff
@@ -0,0 +1,13 @@
+// TODO: define generic gutter values for all components (TA)
+export const ROW_GUTTER = 16
+export const ROW_GUTTER_COMPACT = 12
+export const ROW_SUBHEADING_STYLES = {
```
let's move this into the component itself (see other comment)

mattkrick commented on src/universal/styles/rows.ts:
```diff
@@ -0,0 +1,13 @@
+
+import {PALETTE} from 'universal/styles/paletteV2'
+
+export const ROW_BORDER_COLOR = PALETTE.BORDER.LIGHTER
```
let's remove this & just let components reference `PALETTE.BORDER.LIGHTER`, that way they don't need to import anything, the value just shows up.

mattkrick commented on src/universal/styles/rows.ts:
```diff
@@ -0,0 +1,13 @@
+
+export const ROW_BORDER_COLOR = PALETTE.BORDER.LIGHTER
+// TODO: define generic gutter values for all components (TA)
+export const ROW_GUTTER = 16
```
let's move the gutter values to `constEnums` (maybe under a new GUTTERS enum?)

mattkrick commented on src/universal/styles/ui.js:
```diff
@@ -483,7 +480,7 @@ const ui = {
   // ---------------------------------------------------------------------------
   menuBackgroundColor: '#fff',
-  menuBorderColor: appTheme.palette.mid30l,
+  menuBorderColor: '#DFDFE5',
```
looks like this is only used once. can we remove it from ui.js and replace with BORDER.LIGHTER?

ackernaut commented:
@mattkrick these requests are very helpful! One thing I was navigating: it seems that when a TS files exports the `ENUM` the file importing it _has_ to be either `.ts` or `.tsx` instead of `.js`? I think I’ll have to format some component files to declares types when adding the suggested changes to use `ENUM`. Does that make any sense? I haven’t been in TS land for a second so I need a pointer for sure!

mattkrick commented:
That's right, it's gotta be a typescript file to use enums, so we'll need to change the extension for those older components

ackernaut commented:
@mattkrick I’ve addressed the feedback comments. Sorry this PR got so big with the housekeeping!

Converting these to TS has unleashed some new warnings. I think the RetroTemplateListMenu already had the warning.
- Panel.tsx
- TeamSettings.tsx
- Notifications.tsx
- AddSlackChannel.tsx
- RetroTemplateListMenu.tsx

Could you take a look at those and see if they are easy fixes?

mattkrick commented on src/universal/styles/paletteV2.ts:
```diff
@@ -16,9 +16,11 @@ export namespace PALETTE {
     DARK = '#b6adc7',
-    MAIN = '#493272'
+    MAIN = '#493272',
+    BLUE_RGB = '50, 154, 229'
```
can we just bake the alpha into the hex & use that instead? eg `#329AE580`. That way, the alpha is static so i won't be encouraged to play around with the alpha & make slightly wrong colored blues.

mattkrick commented:
ts warnings fixed

ackernaut commented:
ty looking now!

ackernaut commented:
Ok, ready!

mattkrick commented:
### Test
Everything looks as expected per latest designs/theme:
- [ ] Add provider via card menu
- [ ] Integrate a card via menu
- [ ] Interact with provider list under team settings
- [ ] Interact with provider details under team settings
- [ ] Interact with notifications
- [ ] Interact with organization billing and members
- [ ] Interact with user profile
